server/athenian/api/controllers/miners/github/contributors.py

```diff
@@ -21,14 +21,42 @@ async def mine_contributors(repos: Collection[str],
                             time_to: Optional[datetime],
                             db: databases.Database,
                             cache: Optional[aiomcache.Client],
-                            with_stats: Optional[bool] = True) -> List[dict]:
+                            with_stats: Optional[bool] = True,
+                            with_roles: List[str] = None) -> List[dict]:
     """Discover developers who made any important action in the given repositories and \
     in the given time frame."""
     time_from = time_from or datetime(1970, 1, 1, tzinfo=timezone.utc)
     now = datetime.now(timezone.utc) + timedelta(days=1)
     time_to = time_to or datetime(now.year, now.month, now.day, tzinfo=timezone.utc)
 
-    return await _mine_contributors(repos, time_from, time_to, with_stats, db, cache)
+    if with_roles:
+        fetch_stats = True
+    else:
+        fetch_stats = with_stats
+
+    contribs = await _mine_contributors(repos, time_from, time_to, fetch_stats, db, cache)
+    if not with_roles:
+        return contribs
+
+    # Doing this in post-processing to avoid decreasing cache hits. Otherwise we could
+    # consider caching the different `fetch_*` functions indidually as a inner caching
+    # layer if this becomes a bottleneck.
+    filtered_contribs = []
```

Comment:
ok, we should run isolated queries and ignore the cache here. Some calls are more expensive than the others, we don't expect the webapp to use them now, and a simple cache key based on the user query should perfectly do the job while reducing the complexity. In other words, this is not `PullRequestMiner`.